File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/runtimeTracing/DebugGoTraceTest.java
package opensdk.sdk.apis.klaytnDebug.runtimeTracing;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugGoTraceResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Debug RPC Test")
public class DebugGoTraceTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_goTrace")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String file = "go.trace";
        int seconds = 5;

        DebugGoTraceResponse response = sdk.debug.goTrace(file, seconds).send();
        response.getResult();
    }
}

File: sdk/client/javascript/openapi-test/test/klaytnDebug/runtimeTracing/DebugGoTrace.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_goTrace API', () => {
    test('should return debug_goTrace', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            done();
        };

        const file = "go.trace";
        const seconds = 5;

        sdk.debug.goTrace(file, seconds, {}, callbackOne);
    });
});

